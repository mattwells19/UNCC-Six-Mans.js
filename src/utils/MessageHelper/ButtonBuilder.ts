import { MessageActionRow, MessageButton, MessageOptions } from "discord.js";
import { getEnvVariable } from "../utils";
import CustomButton, { ButtonCustomID, QueueButtonOptions } from "./CustomButtons";

const isDev = getEnvVariable("ENVIRONMENT") === "dev";

const leaveButton = new CustomButton({ customId: ButtonCustomID.LeaveQueue });
const fillTeamButton = new CustomButton({ customId: ButtonCustomID.FillTeam });
const removeAllButton = new CustomButton({ customId: ButtonCustomID.RemoveAll });
const randomTeamsButton = new CustomButton({ customId: ButtonCustomID.CreateRandomTeam });
const chooseTeamsButton = new CustomButton({ customId: ButtonCustomID.ChooseTeam });
const breakMatchButton = new CustomButton({ customId: ButtonCustomID.BreakMatch });
const reportMatchButton = new CustomButton({ customId: ButtonCustomID.ReportMatch });

export default class ButtonBuilder extends MessageButton {
  static queueButtons(options: QueueButtonOptions = { disabled: false }): MessageActionRow {
    const dynamicJoinButton = new CustomButton({
      customId: ButtonCustomID.JoinQueue,
      disabled: options.disabled,
      label: options.disabled && options.buttonId === ButtonCustomID.JoinQueue ? "Please wait..." : undefined,
    });
    const dynamicLeaveButton = new CustomButton({
      customId: ButtonCustomID.LeaveQueue,
      disabled: options.disabled,
      label: options.disabled && options.buttonId === ButtonCustomID.LeaveQueue ? "Please wait..." : undefined,
    });

    const components = [dynamicJoinButton, dynamicLeaveButton];
    if (isDev) {
      components.push(fillTeamButton, removeAllButton);
    }
    return new MessageActionRow({ components: components });
  }

  static fullQueueButtons(): MessageActionRow {
    const components = [chooseTeamsButton, randomTeamsButton, leaveButton];

    if (isDev) {
      components.push(removeAllButton);
    }
    return new MessageActionRow({ components: components });
  }

  static activeMatchButtons(): MessageActionRow {
    const components = [reportMatchButton];
    if (isDev) {
      components.push(breakMatchButton);
    }
    return new MessageActionRow({ components: components });
  }

  static breakMatchButtons(): MessageActionRow {
    return new MessageActionRow({ components: [breakMatchButton] });
  }
}
