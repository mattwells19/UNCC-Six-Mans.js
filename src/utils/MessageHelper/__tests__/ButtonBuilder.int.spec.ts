import ButtonBuilder from "../ButtonBuilder";

jest.mock("../../utils");

describe("Building Buttons", () => {
  it("return queue buttons", () => {
    const result = ButtonBuilder.queueButtons();
    expect(result).toMatchSnapshot();
  });
  it("return full queue buttons", () => {
    const result = ButtonBuilder.fullQueueButtons();
    expect(result).toMatchSnapshot();
  });
  it("return break match buttons", () => {
    const result = ButtonBuilder.breakMatchButtons();
    expect(result).toMatchSnapshot();
  });
  it("return active match buttons", () => {
    const result = ButtonBuilder.activeMatchButtons();
    expect(result).toMatchSnapshot();
  });
  it("return new season buttons", () => {
    const result = ButtonBuilder.newSeasonButtons();
    expect(result).toMatchSnapshot();
  });
});
