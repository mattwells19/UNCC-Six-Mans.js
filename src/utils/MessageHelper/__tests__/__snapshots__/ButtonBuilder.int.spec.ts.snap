// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Building Buttons return active match buttons 1`] = `
Object {
  "components": Array [
    Object {
      "custom_id": "reportMatch",
      "disabled": false,
      "emoji": null,
      "label": "Report Match",
      "style": 3,
      "type": 2,
      "url": null,
    },
  ],
  "type": 1,
}
`;

exports[`Building Buttons return break match buttons 1`] = `
Object {
  "components": Array [
    Object {
      "custom_id": "breakMatch",
      "disabled": false,
      "emoji": null,
      "label": "DEV: Break Match",
      "style": 4,
      "type": 2,
      "url": null,
    },
  ],
  "type": 1,
}
`;

exports[`Building Buttons return full queue buttons 1`] = `
Object {
  "components": Array [
    Object {
      "custom_id": "chooseTeam",
      "disabled": false,
      "emoji": null,
      "label": "Captains",
      "style": 1,
      "type": 2,
      "url": null,
    },
    Object {
      "custom_id": "randomizeTeams",
      "disabled": false,
      "emoji": null,
      "label": "Random",
      "style": 1,
      "type": 2,
      "url": null,
    },
    Object {
      "custom_id": "leaveQueue",
      "disabled": false,
      "emoji": null,
      "label": "Leave",
      "style": 4,
      "type": 2,
      "url": null,
    },
  ],
  "type": 1,
}
`;

exports[`Building Buttons return queue buttons 1`] = `
Object {
  "components": Array [
    Object {
      "custom_id": "joinQueue",
      "disabled": false,
      "emoji": null,
      "label": null,
      "style": 3,
      "type": 2,
      "url": null,
    },
    Object {
      "custom_id": "leaveQueue",
      "disabled": false,
      "emoji": null,
      "label": null,
      "style": 4,
      "type": 2,
      "url": null,
    },
  ],
  "type": 1,
}
`;
